streaming/src/main/scala/org/apache/spark/streaming/ui/AllBatchesTable.scala

```diff
@@ -156,40 +160,232 @@ private[ui] class ActiveBatchTable(
   }
 }
 
-private[ui] class CompletedBatchTable(batches: Seq[BatchUIData], batchInterval: Long)
-  extends BatchTableBase("completed-batches-table", batchInterval) {
+private[ui] class CompletedBatchTableRow(
```

Comment:
I think `CompletedBatchTableRow` isn't required. You can get the data for sorting from `BatchUIData` itself

Reply:
Oh, i missed this. I followed the format form other pages. We can use `BatchUIData` directly.